### PR TITLE
Reorganize workflow authentication and connections

### PR DESCRIPTION
**Pull request by urseberry**
### What does this PR do?
Deletes two pages from Workflows:
- Service Accounts
- Access 

Replaces the above with two new pages:
- Authentication
- Connections

### Motivation
The two new pages represent a more structured, object-oriented view of workflow components. They are better organized than the previous pages and are easier for user to understand.

The new pages include all material provided by the Workflows product team in the following Google Docs:
https://docs.google.com/document/d/1MbQGuT_4-kXoplj1DVfEDlRWU7xc2-dqVOpKluyLYaU/edit#heading=h.25celggmz8pf
https://docs.google.com/document/d/1B43gkgf9yFKMmWp8E_-6aBYgCJjqWf6YaPVnhXCw6Zg/edit#

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by urseberry** [comment]
Editorial review card: https://datadoghq.atlassian.net/browse/DOCS-4658

File: content/en/workflows/access.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+To create a service account for a workflow, you must have either the Datadog admin role, or a custom role with the **Service Account Write** permission. The service account you create adopts your role and permissions. For more information on service accounts and permissions, see [Service accounts][2].
+
+<div class="alert alert-info">The service account you create allows a workflow to be triggered automatically. However, individual actions in the workflow may require <a href="/workflows/connections/">Connections</a>.</div>
+
+### Associate a service account with a workflow
+
+You can dynamically create a service account for your workflow when you [add an automatic trigger][3].
+
+1. Click the **Create** button. The **Create Service Account** dialog opens.
+1. Use the dropdown to select a role for your service account user.
```
Comment (hestonhoffman) [inline content/en/workflows/access.md:41]: We link to service accounts in the section above. I wonder if it's worth linking to [RBAC](https://docs.datadoghq.com/account_management/rbac/?tab=datadogapplication) here, or adding it to related links. That way users can see what 3 of the 4 roles options mean.

File: content/en/workflows/access.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+Credentials and account authentication that you set up in the following Datadog integration tiles automatically propagate to the corresponding actions in Workflows:
+- Jira
+- PagerDuty
+- Slack
+
+Configure the integration tiles by following instructions in [Datadog Integrations][5].
+
+### Connection credentials
+
+Use a custom [connection][6] to authenticate a [generic action][7] or any action for which the integration tile does not offer authentication.The workflow connection object allows you to provide credentials to connect to an integration. The parameters vary by integration.
```
Comment (hestonhoffman) [inline content/en/workflows/access.md:77]:
```suggestion
Use a custom [connection][6] to authenticate a [generic action][7] or any action for which the integration tile does not offer authentication. The workflow connection object allows you to provide credentials to connect to an integration. The parameters vary by integration.
```

File: content/en/workflows/access.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+To connect to an arbitrary service, use the HTTP Connection type, and choose from two authentication options:
```
Comment (hestonhoffman) [inline content/en/workflows/access.md:79]:
```suggestion
To connect to an arbitrary service, use the HTTP Connection type and choose from two authentication options:
```
Bit of a nit, but I don't think you need this comma

File: content/en/workflows/access.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+- Token based authentication
```
Comment (hestonhoffman) [inline content/en/workflows/access.md:80]:
```suggestion
- Token-based authentication
```
We're a little inconsistent with this one in the docs, but I think "token-based" is a compound adjective here.

File: content/en/workflows/access.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+Use [role-based access control (RBAC)][9] to control access to your workflows and connections. To see the list of permissions that apply to workflows and connections, see [Datadog Role Permissions][10].
+
+### Restrict connection use
+
+Set permissions on each connection to limit modifications or restrict their use. The granular permissions include **Viewer**, **Resolver**, and **Editor**.
+
+Viewer
+: Can view
+
+Resolver
```
Comment (hestonhoffman) [inline content/en/workflows/access.md:105]: I wish we had a better definition for "resolve." I've asked in Slack and we can add it later in a clarifying PR if necessary.

File: content/en/workflows/connections.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+1. Select **Delete**.
+
+### Restrict connection use
+ 
+To learn how to restrict connection use, see [Access and Authentication][4].
+
+## HTTP connection
+
+To connect to an arbitrary service, use the HTTP connection type, and choose from two authentication options:
+- Token based authentication
```
Comment (hestonhoffman) [inline content/en/workflows/connections.md:65]:
```suggestion
- Token-based authentication
```

File: content/en/workflows/access.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
+---
+title: Access and Authentication
+kind: documentation
+description: Access and authentication for Workflows
+aliases:
+  - "/workflows/service_accounts"
+  - "/workflows/setup"
```
Comment (hestonhoffman) [inline content/en/workflows/access.md:7]:
```suggestion
  - "/workflows/setup"
```
I wonder if we should move this alias to the connections page seeing as though the content the user is most likely to have bookmarked is on that page? I could see it both ways though.

**Review by hestonhoffman — COMMENTED** [review]
Left some comments for you. This is great! I think the new structure makes way more sense.

File: content/en/workflows/access.md (hunk 1 above)
Comment (urseberry) [reply content/en/workflows/access.md:41]: Agree that users need a link to RBAC somewhere. I think I'll add the link up above, actually, after the link to the Service accounts doc.

File: content/en/workflows/access.md (hunk 7 above)
Comment (urseberry) [reply content/en/workflows/access.md:7]: I see what you mean. The majority of the information that was on the "setup" page is now on "connections." I'll move the alias.

**Review by hestonhoffman — APPROVED** [review]
Sorry, I missed the re-review on this one. This LGTM!

**Pull request merged**